salt/modules/mac_brew_pkg.py

```diff
@@ -93,13 +93,26 @@ def _tap(tap, runas=None):
     return True
 
 
+def _homebrew_os_bin():
+    """
+    Fetch PATH binary brew full path eg: /usr/local/bin/brew (symbolic link)
+    """
+    return salt.utils.path.which("brew")
+
+
 def _homebrew_bin():
     """
-    Returns the full path to the homebrew binary in the PATH
+    Returns the full path to the homebrew binary in the homebrew installation folder
     """
-    ret = __salt__["cmd.run"]("brew --prefix", output_loglevel="trace")
-    ret += "/bin/brew"
-    return ret
+    brew = _homebrew_os_bin()
+    if not brew:
```

Comment:
`salt.utils.path.which("brew")` returns None already if `brew` doesn't already exist. This isn't a hard ask, so you can implement it if you want, but we could continue just returning the result of the command.

Reply:
Hi again @Ch3LL, I think that I understood correctly your suggestion, it's now returning `_homebrew_os_bin()` value directly in case there is no a true value there and in case there is a true value, it's fetching the installation path and overriding the variable to return. I've also rebased the branch again with latest master version.

Please if I misunderstood your suggestion or you have a new one do not hesitate to comment and I'll be happy to apply it.